modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -735,6 +735,34 @@
 				[b]Note:[/b] As annotations describe their subject, the [annotation @tool] annotation must be placed before the class definition and inheritance.
 			</description>
 		</annotation>
+		<annotation name="@tool_button">
+			<return type="void" />
+			<param index="0" name="text" type="String" default="&quot;&quot;" />
+			<param index="1" name="icon" type="StringName" default="&quot;&quot;" />
+			<description>
+				Mark a function to be displayed as a clickable button in the inspector when running the editor. The parameters are optional and can be used to customize the button.
+				If [param text] is specified it is used as the label text of the button. If [param text] is unspecified then the label text will be derived from the name of the annotated function and automatically capitalized.
+				If [param icon] is specified it is used to fetch a suitable icon texture for the button via [method Control.get_theme_icon] by selecting the desired [param icon] from the [code]"EditorIcons"[/code] theme type. If [param icon] is unspecified the default [code]"Callable"[/code] icon will be used instead.
+				The first parameter of the annotated function is also optional and is an [EditorUndoRedoManager]. Consider using the optionally passed [EditorUndoRedoManager] to allow the action to be reverted safely. The code snippet below demonstrates this:
```

Comment:
Technically the code snippet below demonstrates "toot" first so the sentence is a little odd, but I can let that slip.
I believe **EditorUndoRedoManager** is _technically_ a singleton, as there is only one of them at any given time.
```suggestion
				The first argument of the annotated function is optional and is the [EditorUndoRedoManager]. Consider using the passed [EditorUndoRedoManager] to allow the action to be reverted safely:
```